WalletWasabi/WabiSabi/Models/RoundState.cs

```diff
@@ -10,9 +11,17 @@ namespace WalletWasabi.WabiSabi.Models
 		CredentialIssuerParameters AmountCredentialIssuerParameters,
 		CredentialIssuerParameters VsizeCredentialIssuerParameters,
 		FeeRate FeeRate,
+		Phase Phase,
 		MultipartyTransactionState CoinjoinState)
 	{
 		public static RoundState FromRound(Round round) =>
-			new RoundState(round.Id, round.AmountCredentialIssuerParameters, round.VsizeCredentialIssuerParameters, round.FeeRate, round.CoinjoinState);
+			new RoundState(round.Id, round.AmountCredentialIssuerParameters, round.VsizeCredentialIssuerParameters, round.FeeRate, round.Phase, round.CoinjoinState);
+
+		public TState Assert<TState>() where TState : MultipartyTransactionState =>
+			CoinjoinState switch
+			{
+				TState s => s,
+				_ => throw new InvalidOperationException($"{typeof(TState).Name} state was expected but {CoinjoinState.GetType().Name} state was received.")
```

Comment:
why is github angry at the closing paren?

Reply:
I have no idea. The compiler says it is okay.